### ISSUE TITLE
[Bug]: h3 to h6 look like text

### ISSUE THREAD

**Issue by ChristophWurst** [nextcloud/server#43393]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

![Bildschirmfoto vom 2024-02-06 14-25-27](https://github.com/nextcloud/server/assets/1374172/50407f51-7e10-403b-b0aa-bab467b3e0cc)

h2 isn't great, h3 is bad.

Another example:
![image](https://github.com/nextcloud/server/assets/1374172/f5e13f72-4cd6-4422-962b-aee4d50677b8)

*Encryption* is a h3. It doesn't look like one at all. This makes it super hard to recognize the structure of a text page in Nextcloud, like the settings.

Another one:

![image](https://github.com/nextcloud/server/assets/1374172/873f7e3f-f992-4e1f-8ca5-6acf88565131)

*Update channel* looks like ordinary text. It gives no structure except for the increased spacing.

h1 to h6 all share the same base styling: https://github.com/nextcloud/server/blob/375dade49b28d3ca3d670c585c44b1651458b17f/core/css/styles.scss#L19-L30. The font weight of h2 is set to bold. The rest stays *initial*, which is too light IMO. https://github.com/nextcloud/server/blob/375dade49b28d3ca3d670c585c44b1651458b17f/core/css/apps.scss#L60-L78

### Steps to reproduce

1. Open settings, apps, etc

### Expected behavior

Headings that help me recognize the structure of text, paragraps and sections.

### Installation method

None

### Nextcloud Server version

master

### Operating system

None

### PHP engine version

None

### Web server

None

### Database engine version

None

### Is this bug present after an update or on a fresh install?

None

### Are you using the Nextcloud Server Encryption module?

None

### What user-backends are you using?

- [ ] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

**Comment by ChristophWurst** [comment]
![image](https://github.com/nextcloud/server/assets/1374172/0f1a29f2-a41e-40f4-965c-523452d3ff34)

Text doesn't have the issue because there is a custom style for h* inside the prosemirror editor.

![image](https://github.com/nextcloud/server/assets/1374172/26b0df6a-dce9-451f-9db8-9b95f99425df)

Talk has custom styles through nextcloud/vue and the richttexteditable

If we make good defaults, prosemirror in text, ckeditor in Mail, richtexteditable in Talk, etc could all look the same.

**Comment by ChristophWurst** [comment]
@jancborchardt is this something worth standardizing?

**Comment by ChristophWurst** [comment]
Discussed with other engineering leads. The change is simple but the effects are significant and we are close to the feature freeze of 29. We will therefore move this to after the stable29 branch off.

**Comment by susnux** [comment]
> The change is simple but the effects are significant and we are close to the feature freeze of 29.

Sounds reasonable, so merge after branch off?

**Comment by ChristophWurst** [comment]
Yes :)
